### PR TITLE
Display the original name of component.

### PR DESCRIPTION
**Pull request by lonicerae**
* In the old days, the '--summerize' option could show the component's original's name in the svn URI because it's not escaped. After the commit 3dbf4fb0 the component's name would be escaped. So it's not easy to tell which file was modified based on the svn URI format. This small patch appends the original name to the URI result so that the user could read the file name after displaying the svn URI.

### CODE REVIEW

**Comment by MarsrockEarth** [comment]
您的来信已收到

**Review by d-sahlberg — COMMENTED** [review]
Thank you for the patch!

I understand the idea of the patch but I'm not sure I think this implementation is a good idea.

As noted in the code (lines 169-171) the format is the same as svn status. Changing the output format will break any scripts relying on parsing the text output. Of course, that could be acceptable, but the bar should be quite high in my opinion.

The extra column will be output both in svn diff [working copy path] and svn diff [url]. When operating on a wc path, the extra data will be the same (plus leading and trailing single quote). When operating on a URL, the extra data will be the path component of the url, without url encoding.

The new format would be very hard to parse:
- Spllitting on space will not work if the path contains a space (it will be output verbatim at LEAST in the second component).
- Splitting on single quote will not work if the path contains a single qoute.

All in all, I'm -0 to accept this patch: I don't agree but I wouldn't stand in the way of consensus.

**Comment by d-sahlberg** [comment]
Changing the call to svn_path_url_add_component2() to svn_path_join() would remove the url encoding without (otherwise) changing the output format.

svn_path_join is deprecated, so probably not a good idea to call it directly - some additional work would be required to make the _internal function available (remove static and add to a private header).

**Comment by lonicerae** [comment]
Hey @dsahlberg-apache-org ! Thank you so much for reviewing this PR! What you said is exactly what I was concerned about. Such a change would indeed render scripts that rely on command outputs unusable. To be honest, I'm not a seasoned SVN user, and this issue came up during conversations with my friends about their operations challenges. So, I submitted this commit with a "let's give it a try" mindset.

I’d like to hear your thoughts on how we might improve this to partially retain the output format of the old version (i.e., properly displaying filenames that need escaping) while also supporting the current URI-style output. How could we achieve a balance between the two? I understand that removing the --summarize option allows filenames to display correctly, but it would be more helpful if filenames could be included in a single line. Thank you very much!

**Comment by hartmannathan** [comment]
In case you haven't seen it... There is a discussion about this PR on the mailing list (see thread at: [https://lists.apache.org/thread/8y5dbfxkzc1m1g9x5wpmb0kfp6yfdjgb](https://lists.apache.org/thread/8y5dbfxkzc1m1g9x5wpmb0kfp6yfdjgb))

In particular, please see the following message: [https://lists.apache.org/thread/9n51xxtdjmmxrmw4o2l75cbchkpngmjm](https://lists.apache.org/thread/9n51xxtdjmmxrmw4o2l75cbchkpngmjm) which explains how to use a working copy to get what you're asking for ("on-disk paths in local style").

IMHO we should not change the output format due to potential for breakage. Also, other problems are likely, as explained in the above message.

**Comment by lonicerae** [comment]
@hartmannathan @dsahlberg-apache-org @stspdotname  You're so kind! Thank you very much! I should register an account to join the discussion on the mailing list. Let me share your valuable thoughts with my friend. I'll close this PR now.

**Comment by hartmannathan** [comment]
@lonicerae thanks for your input. You're always welcome to join the mailing lists (see [https://subversion.apache.org/mailing-lists.html](https://subversion.apache.org/mailing-lists.html))... Hope to see you there!

**Comment by lonicerae** [comment]
Thanks @hartmannathan ! Just subscribed.;)
